lib/PipelineC/Tracing/Runner.cpp

```diff
@@ -0,0 +1,448 @@
+      Result.assign(Argument.getScalar().begin(), Argument.getScalar().end());
+    }
+    // Zero-terminate the string
+    Result.push_back(0);
+    return Result;
+  }
+
+  template<ConstexprString Name, IntegerType ArgT>
+  ArgT parseArgumentImpl(ArgumentRef Argument) {
+    return Argument.asInt<ArgT>();
+  }
+
+  template<ConstexprString Name, typename ArgT>
+  // clang-format off
+  requires (anyOf<ArgT, char **, const char **>())
+  std::vector<const char *> parseArgumentImpl(ArgumentRef Argument) {
+    // clang-format on
+    return utils::toStringList(Argument);
+  }
+
+  template<ConstexprString Name, typename ArgT>
+  // clang-format off
+  requires (std::is_pointer_v<ArgT> &&
+            isInteger<std::remove_pointer_t<ArgT>>())
+  std::vector<std::remove_pointer_t<ArgT>>
+  parseArgumentImpl(ArgumentRef Argument) {
+    // clang-format on
+    return Argument.toIntList<std::remove_pointer_t<ArgT>>();
+  }
+
+  template<ConstexprString Name, typename ArgT>
+  // clang-format off
+  requires (std::is_pointer_v<ArgT> && isRPType<remove_constptr<ArgT>>())
+  ArgT parseArgumentImpl(ArgumentRef Argument) {
+    // clang-format on
+    revng_check(Argument.isScalar(), "Argument is not scalar");
+    const std::string &PointerName = Argument.getScalar();
+    ArgT Pointer = getPointer<ArgT>(PointerName);
+    if constexpr (isDestroy<Name>())
+      invalidatePointer(PointerName);
+    return Pointer;
+  }
+
+  template<ConstexprString Name, typename ArgT>
+  // clang-format off
+  requires (isList<ArgT>() &&
+            isRPType<remove_constptr<remove_constptr<ArgT>>>())
+  std::vector<std::remove_pointer_t<ArgT>>
+  parseArgumentImpl(ArgumentRef Argument) {
+    // clang-format on
+    revng_check(Argument.isSequence(), "Argument is not a sequence");
+    using ElementT = std::remove_pointer_t<ArgT>;
+    std::vector<ElementT> Result;
+    for (auto &Elem : Argument.getSequence()) {
+      Result.push_back(getPointer<ElementT>(Elem));
+      if constexpr (isDestroy<Name>())
+        invalidatePointer(Elem);
+    }
+
+    utils::poisonTail(Result);
+    return Result;
+  }
+
+  // clang-format off
+public:
+  // clang-format on
+
+  template<ConstexprString Name, typename ArgT, size_t I>
+  ArgStorage<ArgT>
+  parseArgument(ArgumentsRef Arguments) {
+    if constexpr (anyOf<ArgT, char *, const char *>())
+      return parseString<Name, ArgT, I>(Arguments[I]);
+    else
+      return parseArgumentImpl<Name, ArgT>(Arguments[I]);
+  }
+
+  template<ConstexprString Name, typename ArgT, size_t I, typename... Args>
+  ArgT unwrapStorage(std::tuple<Args...> Arguments) {
+    if constexpr (isList<ArgT>()) {
+      return (ArgT) std::get<I>(Arguments).data();
+    } else {
+      if constexpr (anyOf<ArgT, char *, const char *>()) {
+        if constexpr (isDestroy<Name>()) {
+          // _destroy method parameter is not a string, but a pointer to the
+          // string's address
+          ArgStorage<ArgT> Vector = std::get<I>(Arguments);
```

Comment:
```suggestion
          auto Vector = std::get<I>(Arguments);
```

Reply:
No need, the type in that branch is `std::vector<char>`.